apps/system/test/unit/task_manager_test.js

```diff
@@ -1,1377 +1,570 @@
+
+      var startHide = tm.hide();
+      assert.isTrue(tm._isTransitioning);
+      startHide.then(() => {
+        assert.isFalse(tm._isTransitioning);
+        assert.isFalse(tm.isShown());
+        assert.isFalse(isActivated);
+        assert.ok(document.querySelector('#screen:not(.cards-view)'));
+        assert.equal(document.querySelectorAll('.card').length, 0);
         done();
-      }, failOnReject);
-
-      showTaskManager(this.sinon.clock);
-    });
-
-    suite('display empty cardsview >', function() {
-      setup(function() {
-        showTaskManager(this.sinon.clock);
-      });
-
-      test('on click, empty cardsview is closed and back to home screen',
-      function(done) {
-        var events = [];
-        assert.isTrue(cardsView.classList.contains('empty'));
-        assert.isTrue(cardsView.classList.contains('active'));
-        assert.isTrue(taskManager.isShown());
-
-        waitForEvent(window, 'cardviewclosed', 501).then(function(){
-          events.push('cardviewclosed');
-        }, failOnReject).then(function() {
-          assert.equal(events.length, 1, 'sanity check, only 1 event received');
-          assert.equal(events[0],
-                      'cardviewclosed',
-                      'cardviewclosed event raised when touch starts');
-          assert.isFalse(cardsView.classList.contains('active'));
-          assert.isFalse(taskManager.isShown());
-        }, failOnReject)
-        .then(done, done);
-
-        taskManager.handleEvent({target: taskManager.element,
-                              type: 'click',
-                              preventDefault: function(){} });
-        this.sinon.clock.tick(501);
       });
+      clock.tick(1000);
     });
-  });
 
-  suite('hide > ', function() {
-    setup(function() {
-      MockStackManager.mStack = [
-        appConfigs.sms,
-        appConfigs.game
-      ].map((config) => {
-        var app = new AppWindow(config);
-        apps[app.name] = app;
-        return app;
+    test('Should not show if already showing', (done) => {
+      var spyBeforeShow = spyEvent(window, 'cardviewbeforeshow');
+      assert.isTrue(tm.isShown()); // (noting that we're already shown)
+      tm.show().then(() => {
+        assert.isFalse(spyBeforeShow.called);
+        done();
       });
-      MockStackManager.mCurrent = 0;
-      MockService.mockQueryWith('AppWindowManager.getActiveWindow', apps.sms);
-
-      showTaskManager(this.sinon.clock);
     });
 
-    test('taskManager should not be active', function() {
-      taskManager.hide();
-      assert.isFalse(taskManager.isShown(), 'isShown is false');
-      assert.isFalse(cardsView.classList.contains('active'),
-                    'no .active class');
+    test('Should not respond to holdhome if already showing', () => {
+      var showSpy = sinon.spy(tm, 'show');
+      tm.respondToHierarchyEvent(new CustomEvent('holdhome'));
+      assert.isFalse(showSpy.called);
     });
```

Comment:
and for this test, instead of just spying on `show` I'd like to also assert on the active app (homescreen and not lockscreen) to make sure we're calling `close` properly and setting the special transition classes `from-home` correctly